backend/corpora/lambdas/api/v1/dataset.py

```diff
@@ -59,22 +60,26 @@ def get_status(dataset_uuid: str, user: str):
 @db_session()
 def delete_dataset(dataset_uuid: str, user: str):
     """
-    Cancels an inprogress upload.
+    Deletes an existing dataset or cancels an in progress upload.
     """
     dataset = Dataset.get(dataset_uuid)
     if not dataset:
         raise ForbiddenHTTPException()
     if not Collection.if_owner(dataset.collection.id, dataset.collection.visibility, user):
         raise ForbiddenHTTPException()
+    if dataset.collection_visibility == CollectionVisibility.PUBLIC:
+        return make_response(jsonify("Can not delete a public dataset"), 405)
     curr_status = dataset.processing_status
     if curr_status.upload_status is UploadStatus.UPLOADED:
```

Comment:
This will prevent us from ending in a weird situation where we try to delete a dataset that is still in the validation and conversion process.

```suggestion
    if curr_status.processing_status is ProcessingStatus.SUCCESS:
```